clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -2224,6 +2224,28 @@ LogicalResult MemCpyOp::verify() {
   return mlir::success();
 }
 
+//===----------------------------------------------------------------------===//
+// GetMemberOp Definitions
+//===----------------------------------------------------------------------===//
+
+LogicalResult GetMemberOp::verify() {
+
+  const auto recordTy = getAddrTy().getPointee().dyn_cast<StructType>();
+  if (!recordTy)
+    return emitError() << "expected pointer to a record type";
+
+  if (recordTy.getMembers().size() <= getIndex())
```

Comment:
Should we also skip the verifier for now if the struct is incomplete? Like I mentioned in https://github.com/llvm/clangir/issues/231#issuecomment-1679439639, even though this is right, it might break generating an internal use case - so perhaps you add this verifier work around here and remove it once you add support for recursive references?